File: frontend/utils/validators.js
export function isHrMail(email) {
    const emailRegex = new RegExp(String.raw`^[\w\-\.]+@hr\.nl$`);
    return emailRegex.test(email);
}

export function isISBN(isbn) {
    const isbnRegex = new RegExp(String.raw`
^(?:ISBN(?:-1[03])?:? )?(?=[0-9X]{10}$|(?=(?:[0-9]+[- ]){3})
[- 0-9X]{13}$|97[89][0-9]{10}$|(?=(?:[0-9]+[- ]){4})[- 0-9]{17}$)
(?:97[89][- ]?)?[0-9]{1,5}[- ]?[0-9]+[- ]?[0-9]+[- ]?[0-9X]$
`.replaceAll('\n', ''));
    return isbnRegex.test(isbn);
}

File: backend/utils/validators.py
import re


def is_hr_mail(email):
    # Only allow xyz123@hr.nl emails
    regex = r"^[\w\-\.]+@hr\.nl$"
    return re.fullmatch(regex, email) is not None

def is_isbn(isbn):
    regex = r"""^(?:ISBN(?:-1[03])?:? )?(?=[0-9X]{10}$|(?=(?:[0-9]+[- ]){3})
[- 0-9X]{13}$|97[89][0-9]{10}$|(?=(?:[0-9]+[- ]){4})[- 0-9]{17}$)
(?:97[89][- ]?)?[0-9]{1,5}[- ]?[0-9]+[- ]?[0-9]+[- ]?[0-9X]$""".replace('\n', '')
    return re.fullmatch(regex, isbn) is not None